src/Xrefcheck/Util/Interpolate.hs

```diff
@@ -0,0 +1,54 @@
+{- SPDX-FileCopyrightText: 2018-2019 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+
+module Xrefcheck.Util.Interpolate where
+
+import Universum
+
+import Data.Text.Lazy qualified as TL
+import Data.Text.Lazy.Builder (fromLazyText, toLazyText)
+import Fmt (Buildable, Builder, blockListF, blockListF', indentF)
+
+{-# HLINT ignore "Avoid functions that generate extra trailing newlines/whitespaces" #-}
+
+-- | Like @Fmt.indentF@, but strips trailing spaces and do not adds newline.
+--
+-- >>> import Fmt
+-- >>> indentF 2 "a\n\nb"
+-- "  a\n  \n  b\n"
+--
+-- >>> interpolateIndentF 2 "a\n\nb"
+-- "  a\n\n  b"
+interpolateIndentF :: Int -> Builder -> Builder
+interpolateIndentF n b = (case TL.last (toLazyText b) of
+  '\n' -> id
+  _ ->  stripLastNewline) $ stripTrailingSpaces $ indentF n b
+  -- strips newline added by indentF
+
+-- | Like @Fmt.blockListF'@, but strips trailing spaces and do not adds newline
+interpolateBlockListF' :: Text -> (a -> Builder) -> NonEmpty a -> Builder
+interpolateBlockListF' =  stripLastNewline . stripTrailingSpaces ... blockListF'
+
+-- | Like @Fmt.blockListF@, but strips trailing spaces and do not adds newline
+interpolateBlockListF :: Buildable a => NonEmpty a -> Builder
+interpolateBlockListF = stripLastNewline . stripTrailingSpaces . blockListF
+
+-- remove trailing whitespace from all lines.
+-- Note: output always ends with newline (adds trailing newline if there wasn't one).
+stripTrailingSpaces :: Builder -> Builder
+stripTrailingSpaces
+  = fromLazyText
+  . TL.unlines
+  . map (TL.stripEnd)
+  . TL.lines
+  . toLazyText
+
+stripLastNewline :: HasCallStack => Builder -> Builder
```

Comment:
We should also add `HasCallStack` to `interpolateIndentF`, `interpolateBlockListF` and `interpolateBlockListF'`